src/PKSim.Core/Snapshots/Mappers/CompoundMapper.cs

```diff
@@ -32,53 +33,55 @@ public class CompoundMapper : ParameterContainerSnapshotMapperBase<ModelCompound
          _compoundFactory = compoundFactory;
       }
 
-      public override SnapshotCompound MapToSnapshot(ModelCompound compound)
+      public override async Task<SnapshotCompound> MapToSnapshot(ModelCompound compound)
       {
-         return SnapshotFrom(compound, snapshot =>
-         {
-            snapshot.CalculationMethods = _calculationMethodCacheMapper.MapToSnapshot(compound.CalculationMethodCache);
-
-            snapshot.Lipophilicity = mapAlternatives(compound, COMPOUND_LIPOPHILICITY);
-            snapshot.FractionUnbound = mapAlternatives(compound, COMPOUND_FRACTION_UNBOUND);
```

Comment:
Calling await from withing the action works but results in loss of stack trace when catching exceptions or results in exceptiuon being swollen. The way to make it work is to do all the await .... after the SnapshotFrom method